### PR TITLE
dnsdist 1.2.0 (new formula)

### PR DESCRIPTION
**Pull request by Habbie**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/dnsdist.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class Dnsdist < Formula
+  desc "Highly DNS-, DoS- and abuse-aware loadbalancer"
+  homepage "https://www.dnsdist.org/"
+  url "https://downloads.powerdns.com/releases/dnsdist-1.2.0.tar.bz2"
+  sha256 "9885c9ee8ac7076aede586ea58d4642eb877e7b2d76c902254494e2a5a5faa78"
+
+  depends_on "pkg-config" => :build
+  depends_on "lua"
```
Comment (ilovezfs) [inline Formula/dnsdist.rb]: alphabetize the non-build deps amongst themselves

File: Formula/dnsdist.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  depends_on "libedit"
```
Comment (ilovezfs) [inline Formula/dnsdist.rb]: macOS comes with libedit

File: Formula/dnsdist.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  depends_on "boost"
+
+  def install
+    ENV.cxx11
+
+    ENV["SDKROOT"] = MacOS.sdk_path if MacOS.version == :sierra
```
Comment (ilovezfs) [inline Formula/dnsdist.rb]: what error?

File: Formula/dnsdist.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--without-net-snmp",
+                          "--sysconfdir=#{etc}/dnsdist"
+
```
Comment (ilovezfs) [inline Formula/dnsdist.rb]: no need for blank line

File: Formula/dnsdist.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    system "make", "install"
+  end
+
+  test do
+    output = shell_output("#{bin}/dnsdist --version 2>&1")
```
Comment (ilovezfs) [inline Formula/dnsdist.rb]: test should exercise & verify the functionality of the software

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/dnsdist.rb (hunk 2 above)
Comment (Habbie) [reply Formula/dnsdist.rb]: do you have hints on how to use it with autotooled projects? My personal mac has a .pc file but I understand others do not and autoconf is not picking it up on Jenkins.

Comment (ilovezfs) [reply Formula/dnsdist.rb]:
```
iMac-TMP:~ joe$ brew linkage dnsdist
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libc++.1.dylib
  /usr/lib/libedit.3.dylib
  /usr/lib/libncurses.5.4.dylib
Homebrew libraries:
  /usr/local/opt/lua/lib/liblua.5.2.dylib (lua)
Dependencies with no linkage:
  boost
iMac-TMP:~ joe$ 
```
seems to work no?

Is boost build time only?

Comment (Habbie) [reply Formula/dnsdist.rb]: this dep fails only on High Sierra on your Jenkins

Comment (ilovezfs) [reply Formula/dnsdist.rb]: lol looks like apple removed the .pc for libedit on 10.13 … joy

File: Formula/dnsdist.rb (hunk 3 above)
Comment (Habbie) [reply Formula/dnsdist.rb]: comment added. Note that the pdns formula has the same line, this is not a coincidence.

File: Formula/dnsdist.rb (hunk 2 above)
Comment (ilovezfs) [reply Formula/dnsdist.rb]: something like
```
    sdk = MacOS::CLT.installed? ? "" : MacOS.sdk_path
    ENV["LIBEDIT_CFLAGS"] = "-I#{sdk}/usr/include -I#{sdk}/usr/include/editline"
    ENV["LIBEDIT_LIBS"] = "-L/usr/lib -ledit -lcurses"
```

Comment (Habbie) [reply Formula/dnsdist.rb]: > Is boost build time only?

yes, will fix

File: Formula/dnsdist.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
+class Dnsdist < Formula
+  desc "Highly DNS-, DoS- and abuse-aware loadbalancer"
+  homepage "https://www.dnsdist.org/"
+  url "https://downloads.powerdns.com/releases/dnsdist-1.2.0.tar.bz2"
+  sha256 "9885c9ee8ac7076aede586ea58d4642eb877e7b2d76c902254494e2a5a5faa78"
+
+  depends_on "pkg-config" => :build
+  depends_on "lua"
+  depends_on "boost" => :build
```
Comment (ilovezfs) [inline Formula/dnsdist.rb]: should be above pkg-config

File: Formula/dnsdist.rb (hunk 4 above)
Comment (Habbie) [reply Formula/dnsdist.rb]: gone

File: Formula/dnsdist.rb (hunk 6 above)
Comment (Habbie) [reply Formula/dnsdist.rb]: done

File: Formula/dnsdist.rb (hunk 5 above)
Comment (Habbie) [reply Formula/dnsdist.rb]: replaced with a test that verifies that the (Lua) config parser works correctly

File: Formula/dnsdist.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+                          "--without-net-snmp",
+                          "--sysconfdir=#{etc}/dnsdist"
+    system "make", "install"
+  end
+
+  test do
+    conffile = testpath/"dnsdist.conf"
+    conffile.write("setLocal('127.0.0.1')")
+    output = shell_output("#{bin}/dnsdist -C #{conffile} --check-config 2>&1")
+    assert_match "Configuration '#{conffile}' OK!", output
```
Comment (ilovezfs) [inline Formula/dnsdist.rb]:
```
    assert_equal "Configuration '#{conffile}' OK!", output.chomp
```

Comment (ilovezfs) [reply Formula/dnsdist.rb]:
```
    (testpath/"dnsdist.conf").write "setLocal('127.0.0.1')"
    output = shell_output("#{bin}/dnsdist -C dnsdist.conf --check-config 2>&1")
    assert_equal "Configuration 'dnsdist.conf' OK!", output.chomp
```

File: Formula/dnsdist.rb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  homepage "https://www.dnsdist.org/"
+  url "https://downloads.powerdns.com/releases/dnsdist-1.2.0.tar.bz2"
+  sha256 "9885c9ee8ac7076aede586ea58d4642eb877e7b2d76c902254494e2a5a5faa78"
+
+  depends_on "boost" => :build
+  depends_on "pkg-config" => :build
+  depends_on "lua"
+
+  def install
+    ENV.cxx11
```
Comment (ilovezfs) [inline Formula/dnsdist.rb]: does build fail without this?

Comment (ilovezfs) [reply Formula/dnsdist.rb]: ping

File: Formula/dnsdist.rb (hunk 7 above)
Comment (Habbie) [reply Formula/dnsdist.rb]: :+1:

**Comment by ilovezfs** [comment]
:green_apple:

File: Formula/dnsdist.rb (hunk 8 above)
Comment (Habbie) [reply Formula/dnsdist.rb]: apparently not!

**Comment by Habbie** [comment]
I'm ready to squash if you are, @ilovezfs (assuming you need a squash from me to commit this)!

**Comment by ilovezfs** [comment]
:+1:

**Review by ilovezfs — APPROVED** [review]

**Comment by Habbie** [comment]
squashed

**Comment by ilovezfs** [comment]
Thanks for the :new: formula @Habbie! :rocket: :sparkle:

**Pull request closed without merging**